lib/foreman/renderer.rb

```diff
@@ -57,6 +59,11 @@ def foreman_url(action = "built")
               :token     => (@host.token.value unless @host.try(:token).nil?), :kind => action
     end
 
+    def foreman_server_fqdn
+      config = URI.parse(Setting[:unattended_url])
+      config.host || request.host
```

Comment:
I'd much prefer to provide a helper that simply returns the URL rather than have users write against specific setting names, because it provides a good layer of abstraction against internal changes and features.  (For example, foreman_url supports template proxies, which is more complex than just checking the value of a setting.)